conda/recipes/mantid-developer/recipe.yaml

```
@@ -0,0 +1,120 @@
+  run:
+    - ccache
+    - cmake ${{ cmake }}
+    - conda-verify
+    - if: win
+      then: conda-wrappers
+    - doxygen ${{ doxygen }}
+    - eigen ${{ eigen }}
+    - euphonic ${{ euphonic }}
+    - graphviz ${{ graphviz }}
+    - gsl ${{ gsl }}
+    - h5py
+    - hdf4 ${{ hdf4 }}
+    - hdf5 ${{ hdf5 }}
+    - if: linux
+      then: jemalloc ${{ jemalloc }}
+    - jsoncpp ${{ jsoncpp }}
+    - if: win
+      then: lib3mf
+    - libboost-devel ${{ libboost_devel }}
+    - libboost-python-devel ${{ libboost_python_devel }}
+    - if: osx or linux
+      then: libopenblas ${{ libopenblas }}
+    - librdkafka ${{ librdkafka }}
+    - if: linux
+      then: lz4
+    - matplotlib ${{ matplotlib }}
+    - muparser ${{ muparser }}
+    - ninja ${{ ninja }}
+    - numpy>=2.0,<2.2  # This is intentionally different to conda_build_config.yaml, will address later.
+    - occt ${{ occt }}
+    - pip ${{ pip }}
+    - poco ${{ poco }}
+    - psutil ${{ psutil }}
+    - pycifrw
+    - pydantic ${{ pydantic }}
+    - pyqt ${{ pyqt }}
+    - pyqtwebengine
+    - if: linux
+      then: pystack
+    - python-dateutil ${{ python_dateutil }}
+    - python ${{ python }}
+    - if: osx
+      then: python.app
+    - pyyaml ${{ pyyaml }}
+    - qscintilla2 ${{ qscintilla2 }}
+    - qt ${{ qt_main }}
+    - qtconsole ${{ qtconsole }}
+    - qtpy ${{ qtpy }}
+    - if: linux
+      then: qt-gtk-platformtheme
+    - quasielasticbayes ${{ quasielasticbayes }}
+    - quickbayes ${{ quickbayes }}
+    - rattler-build
+    - requests>=2.32.4
+    - scipy ${{ scipy }}
+    - setuptools
+    - sphinx ${{ sphinx }}
+    - sphinx_bootstrap_theme ${{ sphinx_bootstrap_theme }}
+    - tbb-devel ${{ tbb }}
+    - if: linux
+      then: texlive-core ${{ texlive_core }}
+    - if: osx and not arm64
+      then: texlive-core ${{ texlive_core }}
+    - toml ${{ toml }}
+    - versioningit ${{ versioningit }}
+    - zlib
+    - joblib
+    - orsopy ${{ orsopy }}
+    # Not Windows, OpenGL implementation:
+    - if: linux
+      then:
+        - libgl-devel
+        - gxx_linux-64 ${{ cxx_compiler_version }}
+        - libglu ${{ libglu }}
+        # Needed for test suite on Linux
+        - xorg-libxcomposite>=0.4.5
+        - xorg-libxcursor>=1.2.0
+        - xorg-libxdamage>=1.1.5
+        - xorg-libxi>=1.7.10
+        - xorg-libxscrnsaver>=1.2.3
+        - xorg-libxtst>=1.2.3
+    # Development tooling
+    - cppcheck==2.17.1
+    - if: linux
+      then: gcovr>=4.2
+    - pre-commit>=4.0
+    - if: osx
+      then: libcxx
+    # Use conda version of clang to get matching CXXFLAGS.
+    # Pinned to same version as conda forge: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml
+    - if: osx and not arm64
+      then:
+        - clang_osx-64 ${{ c_compiler_version }}
+        - clangxx_osx-64 ${{ cxx_compiler_version }}
+        - clang_osx-arm64 ${{ c_compiler_version }}
+        - clangxx_osx-arm64 ${{ cxx_compiler_version }}
```

These need to be in a separate if block, otherwise it's using the system compiler, which is casuing the macOS failure
```suggestion
    - if: osx and arm64
      then:
        - clang_osx-arm64 ${{ c_compiler_version }}
        - clangxx_osx-arm64 ${{ cxx_compiler_version }}
```